packages/sdk/src/across/clients/bridgePool.ts

```diff
@@ -234,14 +236,19 @@ function joinUserState(
     .mul(userState.balanceOf)
     .div(fixedPointAdjustment);
   const totalDeposited = BigNumber.from(eventState.tokens[userState.address] || "0");
-  const feesEarned = positionValue.sub(totalDeposited);
+  const totalTransferred = BigNumber.from(eventState.lpTransfers[userState.address] || "0");
+  const totalTransferredValue = BigNumber.from(poolState.exchangeRateCurrent)
+    .mul(totalTransferred)
+    .div(fixedPointAdjustment);
```

Comment:
I don't think this will work for more complex scenarios since you can redeem LP tokens that were transferred in and you can transfer out tokens that were deposited. This could result in totalDeposited or totalTransferredValue being negative and could likely cause fees to go negative.

I think this breaks because of the faulty underlying assumption that the transferred value is as of "today's value" but the deposit values are valued as of that point in time.

Consider what would happen if I had a deposit 6 months ago and immediately transferred the tokens out. My deposit was worth $100, but since my tokens that I transferred were valued at _today's value_, they would be valued higher, let's say $115. If you subtract these two, you end up getting a negative fee value.

Reply:
yea i think you are right. i just worked through an example. its because the redeem value implicitly uses an older exchange rate, while the transfer value uses a higher rate.